### ISSUE TITLE
When trying to access project with drf-yasg from amazon api gateway, it always shows json file instead of UI

### ISSUE THREAD

**Issue by jaydeep987** [axnsan12/drf-yasg#611]
Hi!

I have setup my drf project and deployed in EC2 container of aws. Then I enabled aws load balancer and I can access my ec2 machine using load balancer dns for example: http://my-load-balancer-dns.com/pyapi/swagger
And it works fine, it shows swagger ui as expected.

Now I configured amazon API Gateway and point that to that load balancer elb. when I try to access thru api gateway, now instead of swagger Ui, it always gives swagger json in output.
I tried lot of ways but could not figure out what's happening. Is that problem from api gateway or I need some setting in drf like set some header or something ?

I wonder how your testapp is working with heroku. may be it is not using api gateway?

I appreciate any help!

**Comment by noobExtendsBot** [comment]
I uploaded my project on digital ocean, and I am getting 404, in gunicorn log. As usual it running fine on local host, some extra configs are required in production env?

**Comment by jaydeep987** [comment]
Ok I solved mystery! After tons of tries and looking here and there, I found there was problem in API Gateway Request header setting.

Actually drf-yasg also kind of weird let me tell you why. 

After setting up urls as I shown in first image, if you try to access `http://localhost:8000/pyapi/swagger/` it shows UI perfectly. At that time value of request header is `Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9`

Now same URL if you pass request header `"Accept: application/json"` then istead of showing html UI, it shows swagger JSON! wutt!! 🤯

That I found in Amazon API Gateways's test method's output. It was by default sending `"Accept: application/json"` and thats why I was always getting swagger.json in output. That was showstopper thing! 

I changed it to `Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9` and now I could see UI perfectly!

I hope this will save time of many other people like me who are new to this kind of stuff!
